profiler/upload_test.go

```diff
@@ -43,9 +43,10 @@ var testBatch = batch{
 }
 
 func TestTryUpload(t *testing.T) {
-	// Force an empty containerid on this test.
-	defer func(cid string) { containerID = cid }(containerID)
+	// Force an empty containerid and entityID on this test.
+	defer func(cid, eid string) { containerID = cid; entityID = eid }(containerID, entityID)
 	containerID = ""
+	entityID = ""
```

Comment:
There's no assertion that the entity ID is empty. If I change this line to something non-empty the test still passes.

I'm not actually sure why this test asserts an empty container ID in the first place (which I assume you're following for consistency) but it doesn't _seem_ to be necessary. Do you think it's useful? If so let's comment why, otherwise I'd suggest dropping the change to this test.

Reply:
Nice catch, I added the assertion to verify that the entityID is empty.

I am not familiar enough with this code base to know why it was there in the first place. Perhaps it is there to make sure that the functions called won't initialize the container-id or the entity-id. This way, if we refactor the code we will avoid introducing unexpected behaviors.

Reply:
💡 Thought about it a little more, I think this is useful since it asserts that we won't have a header if there is no container ID or entity ID found, which seems to be the desired behavior. So, adding the assertion makes sense. Thanks for updating the test!